### PR TITLE
More psi4 fixes

### PR DESCRIPTION
**Pull request by alesgenova**

### CODE REVIEW

File: cclib/parser/psi4parser.py [hunk 1]
```diff
@@ -115,7 +122,7 @@ def extract(self, inputfile, line):
         #           C          1.415253322400    -0.230221785400     0.000000000000
         # ...
         #
-        if (self.section == "Geometry") and ("Geometry (in Angstrom), charge" in line):
+        if (self.section == "Geometry") and ("Geometry (in Angstrom), charge" in line) and not hasattr(self, 'finite_difference'):
```
Comment (berquist) [inline cclib/parser/psi4parser.py]: Can you put the `and not ...` on a new line?

File: cclib/parser/psi4parser.py [hunk 2]
```diff
@@ -174,7 +181,7 @@ def extract(self, inputfile, line):
             self.set_attribute('mult', mult)
 
         # The printout for Psi4 has a more obvious trigger for the SCF parameter printout.
-        if (self.section == "Algorithm") and (line.strip() == "==> Algorithm <=="):
+        if (self.section == "Algorithm") and (line.strip() == "==> Algorithm <==") and not hasattr(self, 'finite_difference'):
```
Comment (berquist) [inline cclib/parser/psi4parser.py]: Same as above.

File: cclib/parser/psi4parser.py [hunk 3]
```diff
@@ -371,7 +378,7 @@ def get_normalization_factor(exp, lx, ly, lz):
         # repulsion integrals. In that case, there are actually two convergence cycles performed,
         # one for the density-fitted algorithm and one for the exact one, and the iterations are
         # printed in two blocks separated by some set-up information.
-        if (self.section == "Iterations") and (line.strip() == "==> Iterations <=="):
+        if (self.section == "Iterations") and (line.strip() == "==> Iterations <==")  and not hasattr(self, 'finite_difference'):
```
Comment (berquist) [inline cclib/parser/psi4parser.py]: Same as above.

File: cclib/parser/psi4parser.py [hunk 4]
```diff
@@ -654,7 +662,7 @@ def get_normalization_factor(exp, lx, ly, lz):
         #      2    -379.77675264   -7.79e-03      1.88e-02      4.37e-03 o    2.29e-02      6.76e-03 o  ~
         #  ---------------------------------------------------------------------------------------------
         #
-        if (self.section == "Convergence Check") and line.strip() == "==> Convergence Check <==":
+        if (self.section == "Convergence Check") and line.strip() == "==> Convergence Check <=="  and not hasattr(self, 'finite_difference'):
```
Comment (berquist) [inline cclib/parser/psi4parser.py]: Same as above.

File: cclib/parser/psi4parser.py [hunk 5]
```diff
@@ -102,6 +102,13 @@ def extract(self, inputfile, line):
             if self.reference[0] == 'C':
                 self.reference = self.reference[1:]
 
+        # Parse the XC density functional
```
Comment (berquist) [inline cclib/parser/psi4parser.py]: Are you using `metadata` for anything? If so, would be willing to offer us input on what you'd like it to contain? We have multiple running issues: https://github.com/cclib/cclib/issues/451, https://github.com/cclib/cclib/issues/523, https://github.com/cclib/cclib/pull/599.

**Comment by langner** [comment]
Are there any additional logfiles you could contribute to test this code in regressions?

**Comment by alesgenova** [comment]
@berquist 
I addressed your comments, it looks like travis is passing again.

@langner 
I am using some fairly basic input/output at the moment, but I will contribute some additional outputs in the future (together with some proper checks on the values of parsed output)

**Comment by langner** [comment]
> @berquist
> I addressed your comments, it looks like travis is passing again.

Looks good. If you're up for it, please rebase that commit with a more useful message.

> @langner
> I am using some fairly basic input/output at the moment, but I will contribute some additional outputs in the future (together with some proper checks on the values of parsed output)

I understand. If that basic output is not a unit test logfile from our repo, it would be awesome if you could donate it to our regression suite. Even simple example outputs are great for making sure bugs don't come back.

**Comment by alesgenova** [comment]
@langner 
Reworded the commit message!

**Pull request merged**